tidy3d/plugins/design/method.py

```diff
@@ -23,150 +29,88 @@ class Method(Tidy3dBaseModel, ABC):
     name: str = pd.Field(None, title="Name", description="Optional name for the sweep method.")
 
     @abstractmethod
-    def run(self, parameters: Tuple[ParameterType, ...], fn: Callable) -> Tuple[Any]:
+    def run(self, parameters: Tuple[ParameterType, ...], run_fn: Callable) -> Tuple[Any]:
         """Defines the search algorithm (sequential)."""
 
-    @abstractmethod
-    def sample(self, parameters: Tuple[ParameterType, ...], **kwargs) -> Dict[str, Any]:
-        """Defines how the design parameters are sampled."""
+    def _force_int(self, next_point: dict, parameters: list):
+        """Convert a float asigned to an int parameter to be an int. Update dict in place."""
 
-    @staticmethod
-    def assert_hashable(fn_args: dict) -> None:
-        """Raise error if the function arguments aren't hashable (do before computation)."""
-        fn_args_tuple = tuple(fn_args)
-        try:
-            hash(fn_args_tuple)
-        except TypeError:
-            raise ValueError(
-                "Function arguments must be hashable. "
-                "Parameter sweep tool won't work with sets of lists, dicts or numpy arrays. "
-                "Convert these to 'tuple' for a workaround."
-            )
+        for param in parameters:
+            if type(param) == tdd.ParameterInt:
```

Comment:
maybe `isinstance(param, tdd.ParameterInt)` is better, in class we subclass later.